cli/tests/unit_node/dgram_test.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+import { assertEquals } from "../../../test_util/std/assert/mod.ts";
+import { execCode } from "../unit/test_util.ts";
+import { createSocket } from "node:dgram";
+
+const listenPort = 4503;
+const listenPort2 = 4504;
+
+Deno.test("[node/dgram] udp ref and unref", {
+  permissions: { read: true, run: true, net: true },
+}, async () => {
+  const { promise, resolve } = Promise.withResolvers<Uint8Array>();
+
+  const udpSocket = createSocket("udp4");
+  udpSocket.bind(listenPort);
+
+  udpSocket.unref();
+  udpSocket.ref();
+
+  udpSocket.on("message", (buffer, _rinfo) => {
+    resolve(Uint8Array.from(buffer));
+  });
```

Comment:
```suggestion
  let data;
  udpSocket.on("message", (buffer, _rinfo) => {
   data = Uint8Array.from(buffer);
   udpSocket.close();
  });
  udpSocket.on("close", () => {
    resolve();
  });
```

Socket needs to be closed.